testsuite/rabbitmq/pytest_plugin.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+from . import classes
+from . import service
+
+
+def pytest_addoption(parser):
+    group = parser.getgroup("rabbitmq")
+    group.addoption("--rabbitmq")
+    group.addoption(
+        "--no-rabbitmq",
+        help="Disable use of RabbitMQ",
+        action="store_true",
+    )
+
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers",
+        "rabbitmq: per-test RabbitMQ initialization",
+    )
+
+
+def pytest_service_register(register_service):
+    register_service("rabbitmq", service.create_rabbitmq_service)
+
+
+@pytest.fixture
+def rabbitmq(
+    _rabbitmq,
+) -> classes.Control:
+    return _rabbitmq
+
+
+@pytest.fixture
+def _rabbitmq(
+    _rabbitmq_service,
+    _rabbitmq_service_settings,
+) -> classes.Control:
+    return classes.Control(
```

Comment:
is rabbitmq connection created on each test? it's better to have it established once per session.

Reply:
Good point, that somehow escaped my train of thoughts